### PR TITLE
fix(webpack): avoid grouping with default name

### LINKED ISSUE #7795: [2.14.0] Async components don't get separate bundles
# Versions

- nuxt: v2.14.0
- node: v12.18.1

### Reproduction

https://github.com/danielroe/webpack-async

### Steps to reproduce

1. Clone repo
2. Run `yarn build --analyze` - note that there is a separate async bundle for the huge dependency `twilio-video`
3. Run `yarn add nuxt@2.14 && yarn build --analyze` - note that there is no separate async bundle

### What is expected?

Bundle behaviour should not change in this significant of a way.

### What is actually happening?

Significant performance degradation by bundling all dependencies in just five JS files (even on large sites), meaning we don't benefit from code-splitting or async dependencies.

### Notes
* Regression introduced in b4ffdab790b89b7fe806b2168a70925dd1f5e10b

### PR DESCRIPTION
**Pull request by pi0**
## Types of changes

- [x] Bug fix (a non-breaking change which fixes an issue)
- [ ] New feature (a non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Description

Fixes #7795 by letting webpack to use computed cache group key.

Related:
- #7463 (initial PR)
- #7603
- #7639 (fixing #7625 and #7603 and 7463#issuecomment-650734494)
- 2302ffa40b2e83775ca9f7d5ab7d63b38b523333
- #7667 (fixing #7665)

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly. (PR: #)
- [ ] I have added tests to cover my changes (if not applicable, please state why)
- [ ] All new and existing tests are passing.

### CODE REVIEW

**Comment by pi0** [comment]
I did a bunch of tests to ensure this fix is not causing dot file issue and webpack's default `[name]` generator ([src](https://github.com/webpack/webpack/blob/v4.44.0/lib/optimize/SplitChunksPlugin.js#L162)) returns to fallback into default key still would be nice if we can beta test it on `nuxt-edge`  /cc @danielroe @aldarund @simllll @husayt

**Review by clarkdo — COMMENTED** [review]
Do we have an exmaple about what the new output dirs and files would be ?

**Comment by pi0** [comment]
For #7795 reproduction:

```
.nuxt/dist/client
├── app.97f1f0a.js
├── LICENSES
├── pages
│   └── index.d590496.js
├── runtime.d718bc0.js
└── vendors
    ├── commons.15ff78a.js
    └── pages
        └── index.dba88f0.js
```

With a vendor shared between two pages:

```
├── app.52c70b1.js
├── LICENSES
├── pages
│   ├── foo.56fb624.js
│   └── index.e9174bf.js
├── runtime.b2c0774.js
└── vendors
    ├── commons.4e61828.js
    └── foo~index.f8232e7.js
```

Bud sadly (or happily :D) no simple reproduction for dotfile issue happened in previous releases

**Comment by danielroe** [comment]
I have projects that can replicate both issues so will test.

File: package.json [hunk 1]
```diff
@@ -76,6 +76,7 @@
     "rollup-plugin-license": "^2.1.0",
     "sass-loader": "^8.0.2",
     "sort-package-json": "^1.44.0",
+    "twilio-video": "^2.7.1",
```
Comment (danielroe) [inline package.json:79]: can we replicate fixture without this particular package?

**Comment by simllll** [comment]
Will test it tomorrow!

File: package.json (hunk 1 above)
Comment (pi0) [reply package.json:79]: What do you suggest? Initially added to replicate with similar dep

Comment (danielroe) [reply package.json:79]: I will see if I can find another large dependency already within the Nuxt codebase as I hate the thought of adding unnecessarily to install. 🏋️‍♂️

**Comment by simllll** [comment]
> Will test it tomorrow!

Just tested it with nuxt-next in two of our applications that were affected by the . bug before. Looks good to me! Everything seems to work!

**Comment by pi0** [comment]
Update

```
├── app.37f7ad3.js
├── LICENSES
├── node_modules
│   ├── commons.99cd09c.js
│   └── foo~index.c1279c0.js
├── pages
│   ├── foo.f8abb2f.js
│   └── index.2145558.js
└── runtime.a01e42c.js
````

File: package.json (hunk 1 above)
Comment (pi0) [reply package.json]: https://github.com/nuxt/nuxt.js/pull/7808/commits/e9f60c3b31b314245155d7b5f41fdb6aa891df74

**Review by atinux — APPROVED** [review]

**Pull request merged**